### PR TITLE
Add cancellation task support in ForEachAsync helper

### PR DESCRIPTION
**Pull request by aoller**

### CODE REVIEW

File: VAS.Core/Common/AsyncHelpers.cs [hunk 1]
```diff
@@ -55,18 +56,32 @@ public static Task Return ()
...
+		public static Task ForEachAsync<T> (this IEnumerable<T> source, int dop, Func<T, Task> body, CancellationToken? ct = null)
 		{
 			return Task.WhenAll (
 				Partitioner.Create (source).GetPartitions (dop).Select (partition => {
-					return Task.Run (async () => {
-						using (partition) {
-							while (partition.MoveNext ()) {
-								await body (partition.Current);
+					if (!ct.HasValue) {
+						return Task.Run (async () => {
```
Comment (ylatuya) [inline VAS.Core/Common/AsyncHelpers.cs]: Why do you keep this path at all instaed of using the same with out without cancellation token? If there is no cancellation token create a default one.

File: VAS.Core/Common/AsyncHelpers.cs [hunk 2]
```diff
@@ -55,18 +56,32 @@ public static Task Return ()
...
 							}
-						}
-					});
+						});
+					} else {
+						return Task.Factory.StartNew (async () => {
+							using (partition) {
+								while (partition.MoveNext ()) {
+									await body (partition.Current);
+									if (ct.GetValueOrDefault ().IsCancellationRequested) {
```
Comment (ylatuya) [inline VAS.Core/Common/AsyncHelpers.cs]: Use https://msdn.microsoft.com/en-us/library/system.threading.cancellationtoken.throwifcancellationrequested(v=vs.110).aspx

File: VAS.Core/Common/AsyncHelpers.cs (hunk 1 above)
Comment (aoller) [inline VAS.Core/Common/AsyncHelpers.cs]: Changed

File: VAS.Core/Common/AsyncHelpers.cs (hunk 2 above)
Comment (aoller) [inline VAS.Core/Common/AsyncHelpers.cs]: Changed

**Pull request merged**